src/v/cluster/partition_manager.h

```diff
@@ -31,13 +32,7 @@ class partition_manager {
     using ntp_table_container
       = absl::flat_hash_map<model::ntp, ss::lw_shared_ptr<partition>>;
 
-    partition_manager(
-      ss::sharded<storage::api>&,
-      ss::sharded<raft::group_manager>&,
-      ss::sharded<cluster::tx_gateway_frontend>&,
-      ss::sharded<cloud_storage::partition_recovery_manager>&,
-      ss::sharded<cloud_storage::remote>&,
-      ss::sharded<cloud_storage::cache>&);
+    partition_manager(services&);
```

Comment:
Maybe understanding @senior7515's point here? This does feel like it could be a "slippery slope" in enforcing proper (acyclic) dependencies. i.e. "presenting a temptation to violate the proper dependency graph. Is this mitigated by the fact that you still need to explicitly include definitions to actually use other subsystems?

Reply:
to expand on that, I think it would be useful to have a `<module>/services.h` or some other well defined way that doesn't allow a aprogrammer to simply pass `application.h` into all subsystems